### PR TITLE
new rule: star position for generators

### PR DESCRIPTION
**Pull request by xjamundx**
Relates to #1617 and isn't currently ready (no docs yet), but looking for any feedback on the approach.

### CODE REVIEW

File: lib/rules/generator-star.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * @fileoverview Rule to check for the position of the * in your generator functions
+ * @author Jamund Ferguson
+ */
```
Comment (nzakas) [inline lib/rules/generator-star.js]: Add copyright line here.

File: lib/rules/generator-star.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        if (!node.generator) {
+            return;
+        }
+
+        // check for function *name() {}
+        if (position === "end") {
+
+            // * starts where the next identifier begins
+            if (starToken.range[1] !== context.getTokenAfter(starToken).range[0]) {
+                context.report(node, "Expected to see \"function *\".");
```
Comment (nzakas) [inline lib/rules/generator-star.js]: "Expected to see a space before *."

File: lib/rules/generator-star.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }
+
+        // check for function* name() {}
+        if (position === "start") {
+
+            // * ends where the previous identifier ends
+            if (starToken.range[0] !== context.getTokenBefore(starToken).range[1]) {
+                context.report(node, "Expected to see \"function*\".");
```
Comment (nzakas) [inline lib/rules/generator-star.js]: "Expected no space before *."

File: tests/lib/rules/generator-star.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
+/**
+ * @fileoverview Tests for func-style rule.
+ * @author Nicholas C. Zakas
```
Comment (nzakas) [inline tests/lib/rules/generator-star.js]: Update author and add copyright.

**Comment by xjamundx** [comment]
Once #1650 merges I will rebase this thing (so the README doesn't conflict)

File: lib/rules/generator-star.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    var position = context.options[0];
+
+    // TODO: find out the right way to handle this
+    // basically fail silently if they don'"t include a proper position
+    if (["start", "end"].indexOf(position) === -1) {
```
Comment (xjamundx) [inline lib/rules/generator-star.js]: Is this a thing that I need to worry about, or should just assume people will not screw it up?

Comment (nzakas) [inline lib/rules/generator-star.js]: We usually include a logic default so that if someone omits the option it does something useful.

Comment (xjamundx) [inline lib/rules/generator-star.js]: Will default to `end` in the case that nothing is passed, as that's the style you seem to use the most in understanding es6.

**Comment by nzakas** [comment]
Looks good, though looks like your other PR ended up as part of this now?

**Comment by xjamundx** [comment]
Yeah #1650 needs to merge first (because of the structural changes to the README), but I'll rebase this commit once that's done.

**Comment by nzakas** [comment]
Okay merged #1650

**Comment by xjamundx** [comment]
cool i think it's good to go !

**Comment by nzakas** [comment]
LGTM. Thanks.

**Pull request merged**